python/lsst/sims/catUtils/mixins/VariabilityMixin.py

```diff
@@ -386,56 +488,298 @@ def applyBHMicrolens(self, params, expmjd_in):
+                                    + "to get the data")
+
+            _MLT_LC_CACHE = numpy.load(self._mlt_lc_file)
+            _MLT_LC_CACHE_NAME = self._mlt_lc_file
+
+        if not hasattr(self, '_mlt_dust_lookup'):
+            # Construct a look-up table to determine the factor
+            # by which to multiply the flares' flux to account for
+            # dust as a function of E(B-V).  Recall that we are
+            # modeling all MLT flares as 9000K blackbodies.
+
+            if not hasattr(self, 'lsstBandpassDict'):
+                raise RuntimeError('You are asking for MLT dwarf flaring '
+                                   'magnitudes in a catalog that has not '
+                                   'defined lsstBandpassDict.  The MLT '
+                                   'flaring magnitudes model does not know '
+                                   'how to apply dust extinction to the '
+                                   'flares without the member variable '
+                                   'lsstBandpassDict being defined.')
+
+            ebv_grid = numpy.arange(0.0, 7.01, 0.01)
+            bb_wavelen = numpy.arange(200.0, 1500.0, 0.1)
+            hc_over_k = 1.4387e7  # nm*K
+            temp = 9000.0  # black body temperature in Kelvin
+            exp_arg = hc_over_k/(temp*bb_wavelen)
+            exp_term = 1.0/(numpy.exp(exp_arg) - 1.0)
+            ln_exp_term = numpy.log(exp_term)
+
+            log_bb_flambda = -5.0*numpy.log(bb_wavelen) + ln_exp_term
+            bb_flambda = numpy.exp(log_bb_flambda)
+            bb_sed = Sed(wavelen=bb_wavelen, flambda=bb_flambda)
+
+            base_fluxes = self.lsstBandpassDict.fluxListForSed(bb_sed)
+
+            a_x, b_x = bb_sed.setupCCMab()
+            self._mlt_dust_lookup = {}
+            self._mlt_dust_lookup['ebv'] = ebv_grid
+            list_of_bp = self.lsstBandpassDict.keys()
+            for bp in list_of_bp:
+                self._mlt_dust_lookup[bp] = numpy.zeros(len(ebv_grid))
+            for iebv, ebv_val in enumerate(ebv_grid):
+                wv, fl = bb_sed.addCCMDust(a_x, b_x,
+                                           ebv=ebv_val,
+                                           wavelen=bb_wavelen,
+                                           flambda=bb_flambda)
+
+                dusty_bb = Sed(wavelen=wv, flambda=fl)
+                dusty_fluxes = self.lsstBandpassDict.fluxListForSed(dusty_bb)
+                for ibp, bp in enumerate(list_of_bp):
+                    self._mlt_dust_lookup[bp][iebv] = dusty_fluxes[ibp]/base_fluxes[ibp]
+
+        # get the distance to each star in parsecs
+        _au_to_parsec = 1.0/206265.0
+        dd = _au_to_parsec/parallax
+
+        # get the area of the sphere through which the star's energy
+        # is radiating to get to us (in cm^2)
+        _cm_per_parsec = 3.08576e16
+        sphere_area = 4.0*numpy.pi*numpy.power(dd*_cm_per_parsec, 2)
+
+        flux_factor = self.photParams.effarea/sphere_area
+
```

Comment:
Ok, this whole process looks right to me and I'm sure you've already done this, but getting another pair of eyes on the last few lines would probably be good.